examples/plugins/src/file-size.plugin.unit.test.ts

```diff
@@ -0,0 +1,361 @@
+import { vol } from 'memfs';
+import { unlink } from 'node:fs/promises';
+import { basename, join } from 'node:path';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { MEMFS_VOLUME } from '@code-pushup/testing-utils';
+import { formatBytes } from '@code-pushup/utils';
+import {
+  assertFileSize,
+  errorMessage,
+  fileSizeIssues,
+  infoMessage,
+  runnerFunction,
+  scoreFilesizeAudit,
+} from './file-size.plugin';
+
+// Mock file system API's
+vi.mock('fs', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs;
+});
+vi.mock('fs/promises', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs.promises;
+});
```

Comment:
If you merge current `main` to your branch, you can set up shared mocks for the `fs` by adding path `../../testing-utils/src/lib/setup/fs.mock.ts` to `setupFiles` in `vite.config.unit.ts`.